master/internal/provisioner/provisioner.go

```diff
@@ -38,6 +38,7 @@ type Provisioner struct {
 
 type provider interface {
 	instanceType() instanceType
+	setup(ctx *actor.Context)
```

Comment:
we shouldn't have `setup` function because if a provider cannot be created successfully we should just fail at the constructor of the providers rather than on the runtime of `provisioner`

Reply:
We need the `setup` step to build state correctly. `setup` isn't a step that can fail, but it should be executed exactly once when the provisioner is created. Essentially it's just making the `PreStart` actor message available to the provider.